docs/ref/signals.txt

```diff
@@ -546,6 +554,12 @@ Request/response signals
 
 Signals sent by the core framework when processing a request.
 
+.. warning::
+
+    Signals can make your code harder to maintain. Consider :doc:`providing a
```

Comment:
Slightly clearer? Most users aren’t writing third party packages

```suggestion
    Signals can make your code harder to maintain. Consider :doc:`using a
```

Reply:
Agree with this change.